packages/slice-machine/lib/env/npmApi.ts

```diff
@@ -0,0 +1,58 @@
+import npmFetch from "npm-registry-fetch";
+import semver from "semver";
+import type { UpdateVersionInfo } from "../models/common/Environment";
+
+export async function fetchVersions(name: string): Promise<string[]> {
+  return npmFetch
+    .json(name)
+    .then((json) => {
+      const versions = (json.versions || {}) as Record<string, unknown>;
+      return Object.keys(versions);
+    })
+    .catch(() => []);
+}
+
+const highest = (
+  kind: "patch" | "minor" | "major",
+  current: string,
+  versions: string[]
+): string => {
+  const minorVersion = current.replace(/^(\d+\.\d+).*/, "$1");
+  const majorVersion = current.replace(/^(\d+).*/, "$1");
+
+  const top = versions.reduce((acc, version) => {
+    if (/^\d+\.\d+\.\d+$/.test(version) === false) return acc;
+    if (semver.gt(acc, version)) return acc;
+    if (kind === "patch" && version.startsWith(minorVersion)) return version;
+    if (
+      kind === "minor" &&
+      version.startsWith(majorVersion) &&
+      version.startsWith(minorVersion) === false
```

Comment:
Can you simplify this condition with `!version.startsWith(minorVersion)`

Reply:
sure, I think `==false` is more readable, but I'll do it